### PR TITLE
12247: Terminated EEs with active Enrollments 

### PR DESCRIPTION
**Pull request by davidplappert**
|Field|Value|
|-----|-----|
| Redmine Ticket Number | [12247](https://devops.dchbx.org/redmine/issues/12247) |
| App-Dev Road Map # | n/a |
| Start Date | n/a |
| Due Date | n/a  |
| App-Dev Priority | UP-Disqueued |
| Development Story Points | 1 |
| Peer Review Story Points | n/a |
| Ticket Author | @DFos85 |

### CODE REVIEW

File: lib/tasks/terminated_ee_with_active_enrollment_list.rake [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+               enrollment_state
+             )
+      processed_count = 0
+      Dir.mkdir("hbx_report") unless File.exists?("hbx_report")
+      file_name = "#{Rails.root}/hbx_report/terminated_ee_with_active_enrollment.csv"
+      CSV.open(file_name, "w", force_quotes: true) do |csv|
+        csv << field_names
+        census_employees.each do |census_employee|
+          if census_employee.employment_terminated_on<Date.new(2016,10,31)
+            employer=census_employee.employer_role
```
Comment (blackeaglejs) [inline lib/tasks/terminated_ee_with_active_enrollment_list.rake]: What's an employer_role?

File: lib/tasks/terminated_ee_with_active_enrollment_list.rake [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            person=census_employee.employee_role.person
+            benefit_group_assignments=ce.benefit_group_assignments
```
Comment (blackeaglejs) [inline lib/tasks/terminated_ee_with_active_enrollment_list.rake]: Do you mean census_employee? I'm not sure ce is defined anywhere else.

File: lib/tasks/terminated_ee_with_active_enrollment_list.rake (hunk 1 above)
Comment (rabbittracey) [inline lib/tasks/terminated_ee_with_active_enrollment_list.rake]: should be employer_profile

File: lib/tasks/terminated_ee_with_active_enrollment_list.rake (hunk 2 above)
Comment (rabbittracey) [inline lib/tasks/terminated_ee_with_active_enrollment_list.rake]: yes, ce should be census_employee
thanks

File: lib/tasks/terminated_ee_with_active_enrollment_list.rake [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+             )
+      processed_count = 0
+      Dir.mkdir("hbx_report") unless File.exists?("hbx_report")
+      file_name = "#{Rails.root}/hbx_report/terminated_ee_with_active_enrollment.csv"
+      CSV.open(file_name, "w", force_quotes: true) do |csv|
+        csv << field_names
+        census_employees.each do |census_employee|
+          if census_employee.employment_terminated_on<Date.new(2016,10,31)
+            employer=census_employee.employer_profile
+            person=census_employee.employee_role.person
```
Comment (blackeaglejs) [inline lib/tasks/terminated_ee_with_active_enrollment_list.rake]: There could be an instance where the person returns a nil. You should find a way to guard against this.

File: lib/tasks/terminated_ee_with_active_enrollment_list.rake [hunk 4]
```diff
@@ -27,7 +27,7 @@ namespace :report do
         census_employees.each do |census_employee|
           if census_employee.employment_terminated_on<Date.new(2016,10,31)
             employer=census_employee.employer_profile
-            person=census_employee.employee_role.person.present ? census_employee.employee_role.person: nil
+            person=census_employee.try(:employee).try(:person)
```
Comment (blackeaglejs) [inline lib/tasks/terminated_ee_with_active_enrollment_list.rake]: 1) You should avoid using .try if at all possible. 
2) Do you mean employee or employee_role? 
3) Consider asking @arasevic about .maybe method.

File: lib/tasks/terminated_ee_with_active_enrollment_list.rake [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+#the employee has ER-sponsered enrollment not in terminated or cancelled state
+#the date of termination is before 10/31/2016
+require 'csv'
+namespace :report do
+  namespace :user_account do
+    desc "List of terminated employee with active_enrollment"
+    task :employee_list => :environment do
+      census_employees = CensusEmployee.linked.where(aasm_state: "employment_terminated",employee_role_id: {:$exists => true}).all
+      field_names  = %w(
+               hbx_id
```
Comment (blackeaglejs) [inline lib/tasks/terminated_ee_with_active_enrollment_list.rake]: Based on the comments in the ticket, you should consider making this column name "enrollment_hbx_id" to distinguish it from the person's hbx_id.

Comment (blackeaglejs) [inline lib/tasks/terminated_ee_with_active_enrollment_list.rake]: @rabbittracey

File: lib/tasks/terminated_ee_with_active_enrollment_list.rake [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+            benefit_group_assignments=census_employee.benefit_group_assignments
+            #terminate_state=HbxEnrollment::TERMINATED_STATUSES+HbxEnrollment::CANCELED_STATUSES
+            active_state=%w(coverage_selected coverage_enrolled)
+            benefit_group_assignments.each do |benefit_group_assignment|
+              benefit_group_assignment.hbx_enrollments.each do |enrollment|
+                termination_verify=(active_state.include?enrollment.aasm_state)
+                employer_sponsor_verify=(enrollment.kind=="employer_sponsored")
+                if person&&employer_sponsor_verify&& termination_verify
+                    csv << [
+                      hbx.hbx_id,
```
Comment (blackeaglejs) [inline lib/tasks/terminated_ee_with_active_enrollment_list.rake]: are you sure this shouldn't be `enrollment.hbx_id`, not `hbx.hbx_id`? I'm not sure where you define a variable called `hbx`. @rabbittracey

Comment (rabbittracey) [inline lib/tasks/terminated_ee_with_active_enrollment_list.rake]: yes, it should be enrollment.hbx_id

**Pull request closed without merging**